lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -145,11 +159,12 @@ public function bindParam($column, &$variable, $type = null, $length = null)
         $column = isset($this->_paramMap[$column]) ? $this->_paramMap[$column] : $column;
 
         if ($type == \PDO::PARAM_LOB) {
-            $lob = oci_new_descriptor($this->_dbh, OCI_D_LOB);
-            $lob->writeTemporary($variable, OCI_TEMP_BLOB);
+            if (!is_a($variable, 'Oci-Lob')) {
+                throw new OCI8Exception('Can not bind this value to the BLOB column, please use a oci8 descriptor');
+            }
```

Comment:
+1 it's a nice implementation btw I wasn't aware that we could reference this way around to `boundValues`.